sdk/python/feast/repo_config.py

```diff
@@ -290,12 +293,15 @@ def offline_store(self):
     @property
     def auth_config(self):
         if not self._auth:
+            is_oidc_client = "username" in self.auth and "password" in self.auth
             if isinstance(self.auth, Dict):
-                self._auth = get_auth_config_from_type(self.auth.get("type"))(
-                    **self.auth
-                )
+                self._auth = get_auth_config_from_type(
+                    self.auth.get("type"), is_oidc_client
+                )(**self.auth)
             elif isinstance(self.auth, str):
```

Comment:
@tokoko @franciscojavierarceo - I have got this example from online_store config parsing but seems like unnecessary check. Do we really need this check or code block?